core/app/models/spree/product.rb

```diff
@@ -63,6 +63,10 @@ def find_or_build_master
       delegate :"#{attr}", :"#{attr}=", to: :find_or_build_master
     end
 
+    def price
+      master.price || Spree::Price.unscoped { master.price }
```

Comment:
I'm not sure if there is a better way to do this. Basically, if you look at a deleted product in the admin, you probably want to see the price. 

But only get the unscoped version if there isn't a scoped master.price.